### PR TITLE
Fixes for TF and remappings

### PR DESCRIPTION
**Pull request by pavel-kirienko**
#8: TF caching is implemented [as proposed](https://github.com/srv/viso2/issues/8#issuecomment-15403249) by Stephan.
#9: The possibilities to remap a stereo namespace or image topics are gone because this functionality didn't work as expected. Instead, the rather standard concept of [pushing down](http://www.ros.org/wiki/Nodes#Remapping_Arguments.A.22Pushing_Down.22) is recommended. Also please note that the new topic API is compatible with [another VO engines](http://www.ros.org/wiki/vslam_system/Tutorials/RunningVslamOnStereoData#Necessary_Data).

### CODE REVIEW

File: viso2_ros/src/odometer_base.h [hunk 1]
```diff
@@ -187,6 +172,34 @@ class OdometerBase
...
+  {
+    std::string error_msg;
+    if (tf_listener_.canTransform(base_link_frame_id_, sensor_frame_id_, timestamp, &error_msg))
+    {
+      tf_listener_.lookupTransform(base_link_frame_id_,
+          sensor_frame_id_,
+          timestamp,
+          base_to_sensor_);
+    }
+    else if (tf_listener_.canTransform(base_link_frame_id_, sensor_frame_id_, ros::Time(0), &error_msg))
```
Comment (miquelmassot) [inline viso2_ros/src/odometer_base.h:186]: If the last tf is stored in base_to_sensor_ why is this else branch needed?

Comment (pavel-kirienko) [inline viso2_ros/src/odometer_base.h:186]: In some cases TF may fail to extrapolate to a requested timestamp. In such cases, we simply update the cached transform with a last available from the TF listener, assuming that the TF listener may provide the more actual transform than is stored in `base_to_sensor_`.

**Comment by stwirth** [comment]
Thanks @pavel-kirienko for this pull request. You are right, the tf handling in viso2_ros is not very robust. The [solution in fovis_ros](https://github.com/srv/fovis/blob/master/fovis_ros/src/odometer_base.hpp#L137-L141) is better but still not optimal.
I would suggest to do a lookup once on startup using waitForTransform() with a large timeout as this is the most important one. In the callback I would call waitForTransform() with a smaller timeout and use a cached tf if this fails.
I'll try this out now.

**Comment by stwirth** [comment]
I made some changes in the develop branch (see 68ec5469d67167f87db25ad88631cf6cbde6c9c2). If it works fine we'll merge to master.

**Comment by pavel-kirienko** [comment]
Works fine for me.
Why not make `timeout` configurable?

**Pull request closed without merging**